gh-122519: Adding socket module shutdown() constants description (#122543)
python/cpython: Doc/library/socket.rst

   .. versionadded:: 3.12

.. data:: SHUT_RD
          SHUT_WR
          SHUT_RDWR

>    These argument are used to the :func:`socket.shutdown` method of socket object.

Reviewer: ```suggestion
   These constants are used by the :meth:`socket.shutdown` method.
```

You should also check the availability of those constants. They should exist on Linux systems but is there *any* chance of them existing on other systems such as FreeBSD, macOS and Windows?

Author: Thanks for your support and help! I test in Windows 11 by checking if it`s available. And I investigate manual page on different system and it seems all constants are avaialbe.

FreeBSD:
https://man.freebsd.org/cgi/man.cgi?shutdown(2)

MacOS:
https://developer.apple.com/library/archive/documentation/System/Conceptual/ManPages_iPhoneOS/man2/shutdown.2.html

Could you help me check if my method is accurately ? or if there`s any better way to confirm that. I appreciate again your help.

Reviewer: Actually, `shutdown` is available only on non-WASI platforms. So the availability of those constants is `not WASI` (use the same formulation as other constants that are not WASI compatible).

Author: That is, Can I conclude that all traditional operating system(Linux, MacOS, Windows) support these constants ?

Reviewer: Just say "not WASI". No need to enumerate all "traditional" ones. Look at the availability of shutdown() which says "not WASI" and use the same formulation.

Author: Understood, thanks for your help!

   .. versionadded:: 3.12

.. data:: SHUT_RD
          SHUT_WR
          SHUT_RDWR

>    These constants are used by the :meth:`shutdown` method of socket objects.

Reviewer 2: This seems to have lost the hyperlink. Maybe “~socket.socket.shutdown” would do it. See similar instances above, e.g. under SIO_RCVALL.

Author: Let me fix it.

>    These constants are used by the :meth:`socket.shutdown` method of socket object.

Reviewer 2: Fix plural of _objects_, and hide the “socket.” class name prefix:

. . . the _shutdown_ method of socket objects.
